streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -502,56 +494,24 @@ public void closeAndRecycleState() {
         log.info("Closed clean and recycled state");
     }
 
-    private void maybeScheduleCheckpoint() {
-        switch (state()) {
-            case RESTORING:
-            case SUSPENDED:
-                this.checkpoint = checkpointableOffsets();
-
-                break;
-
-            case RUNNING:
-                if (!eosEnabled) {
-                    this.checkpoint = checkpointableOffsets();
-                }
-
-                break;
-
-            case CREATED:
-            case CLOSED:
-                throw new IllegalStateException("Illegal state " + state() + " while scheduling checkpoint for active task " + id);
-
-            default:
-                throw new IllegalStateException("Unknown state " + state() + " while scheduling checkpoint for active task " + id);
-        }
-    }
-
-    private void writeCheckpointIfNeed() {
+    private void maybeWriteCheckpoint() {
```

Comment:
Seems we call `stateMgr.checkpoint` unconditionally now. Should we rename this this `writeCheckpoint` ? Or even remove it all together as we `if (commitNeeded)` check is "just" a guard and the method is a single liner now?

Reply:
Well, the `commitNeeded` guard is a good idea imo. But I agree we should rename this